### PR TITLE
Add options to set different cache values for uploads

### PR DESCRIPTION
**Pull request by leedxw**
- allow static files to have a different max-age
- enable a max-age for non-private responses

- [ ] Version number has been bumped

### CODE REVIEW

File: settings.php [hunk 1]
```diff
@@ -117,18 +117,39 @@ function dxw_members_only_options_page()
...
             <td>
                 <input type="number" min="0" step="1" name="dxw_members_only_max_age" id="dxw_members_only_max_age" value="<?php form_option('dxw_members_only_max_age') ?>" class="regular-text">
                 <span class="description">Defaults to 0 if not set.</span>
             </td>
         </tr>
     </table>
 
+    <table class="form-table">
+        <tr valign="top">
+            <th scope="row"><label for="dxw_members_only_max_age_static"><?php _e('Browser cache max age in seconds (static content)', 'dxwmembersonly') ?></label></th>
```
Comment (snim2) [inline settings.php:135]: I'm not sure this needs to be a separate table, but if I make `Browser cache max age in seconds (static content)` a non-zero number then either the value does not save or it doesn't display correctly on the settings page (0 is displayed)

**Comment by snim2** [comment]
On Saluki (local), I have a private page with a static image, I've change the plugin settings but in the browser I see this header for the image:

```
Cache-control: public, max-age=604800
```

I haven't added `604800` in any of the settings. The header for the page itself is fine `no-cache, must-revalidate, max-age=0, no-store, private`

**Comment by leedxw** [comment]
If you're seeing

```
Cache-control: public, max-age=604800
```

Please double check the nginx config - the following should be disabled 

```
#include /etc/nginx/snippets/wordpress-uploads-public.conf;
```

**Comment by snim2** [comment]
> If you're seeing
> 
> ```
> Cache-control: public, max-age=604800
> ```
> 
> Please double check the nginx config - the following should be disabled
> 
> ```
> #include /etc/nginx/snippets/wordpress-uploads-public.conf;
> ```

Ah, is there an env var I should have set?

**Comment by leedxw** [comment]
```
ENABLE_INTRANET_SETTINGS=true
```

**Comment by snim2** [comment]
@leedxw OK, that seems fine, but not changing/displaying the settings is an issue.

**Comment by snim2** [comment]
@leedxw what counts as "static content"?

**Comment by leedxw** [comment]
Static content is essentially uploads that aren't marked as public.

i.e. where a re-request will almost always be for the exact same resource.

Right now we can't differentiate between a photo and a list of recent news items.

**Review by snim2 — APPROVED** [review]
I think this is working OK now, and the settings certainly save. We will need to bump the version number before we release.

**Pull request merged**